crates/meilisearch/src/main.rs

```diff
@@ -139,6 +170,76 @@ async fn try_main() -> anyhow::Result<()> {
     Ok(())
 }
 
+/// Prompt the user about the contact email for support and news.
+/// It only displays the prompt if the input is an interactive terminal.
+async fn prompt_for_contact_email() -> anyhow::Result<Option<String>> {
+    let stdin = tokio::io::stdin();
+
+    if !stdin.is_terminal() {
+        return Ok(None);
+    }
+
+    println!("Stay up to date! Get monthly updates about new features and tips to get the most out of Meilisearch.");
+    println!("You can use the `--contact-email` parameter to disable this prompt.");
+    print!("Enter your email or leave blank to skip> ");
```

Comment:
I would say that CLI communication should be very heavily skewed towards brevity and raw information. You don't usually fire up a console to have fun and browse around, you're there to accomplish a task.

Introducing the prompt will be an obstacle to that task, and asking for someone's email is particularly invasive and concerning, so a friendly tone is likely to make the experience even more aggravating.

I defer the final decision to you—I'm better at making things better for users, but in this case user experience is significantly less important than the company's needs.

Reply:
You are right, it will not be formatted, so I'll remove this.

```suggestion
    println!("Get monthly updates about new features and tips to get the most out of Meilisearch.");
    println!("Use the --contact-email option to disable this prompt.");
    print!("Enter your email or leave blank to skip> ");
```